core/src/main/scala/com/databricks/labs/remorph/parsers/tsql/TSqlExpressionBuilder.scala

```diff
@@ -224,6 +223,73 @@ class TSqlExpressionBuilder extends TSqlParserBaseVisitor[ir.Expression] with Pa
   }
 
+  // Note that this visitor is made complicated and difficult because the built in ir does not use options
+  // and so we build placeholder values for the optional values. They also do not extend expression
+  // so we can't build them logically with visit and accept. Maybe replace them with
+  // extensions that do do this?
+  override def visitExprOver(ctx: ExprOverContext): ir.Window = {
+    val windowFunction = ctx.expression().accept(this)
+    val partitionByExpressions =
+      Option(ctx.overClause().expression()).map(_.asScala.toList.map(_.accept(this))).getOrElse(List.empty)
+    val orderByExpressions = Option(ctx.overClause().orderByClause())
+      .map(_.orderByExpression().asScala.toList.map { orderByExpr =>
+        val expression = orderByExpr.expression().accept(this)
+        val sortOrder =
+          if (Option(orderByExpr.DESC()).isDefined) ir.DescendingSortDirection
+          else ir.AscendingSortDirection
+        ir.SortOrder(expression, sortOrder, ir.SortNullsUnspecified)
+      })
+      .getOrElse(List.empty)
+
+    val rowRange = Option(ctx.overClause().rowOrRangeClause())
+      .map(buildWindowFrame)
+      .getOrElse(noWindowFrame)
+
+    ir.Window(windowFunction, partitionByExpressions, orderByExpressions, rowRange)
+  }
+
+  private def noWindowFrame: ir.WindowFrame =
+    ir.WindowFrame(
+      ir.UndefinedFrame,
+      ir.FrameBoundary(current_row = false, unbounded = false, ir.Noop),
+      ir.FrameBoundary(current_row = false, unbounded = false, ir.Noop))
+
+  private def buildWindowFrame(ctx: RowOrRangeClauseContext): ir.WindowFrame = {
+    val frameType = buildFrameType(ctx)
+    val bounds = Trees
+      .findAllRuleNodes(ctx, TSqlParser.RULE_windowFrameBound)
+      .asScala
+      .collect { case wfb: WindowFrameBoundContext => wfb }
+      .map(buildFrame)
+
+    val frameStart = bounds.head // Safe due to the nature of window frames always having at least a start bound
+    val frameEnd =
+      bounds.drop(1).headOption.getOrElse(ir.FrameBoundary(current_row = false, unbounded = false, ir.Noop))
+
+    ir.WindowFrame(frameType, frameStart, frameEnd)
+  }
+
+  private def buildFrameType(ctx: RowOrRangeClauseContext): ir.FrameType = {
+    if (Option(ctx.ROWS()).isDefined) ir.RowsFrame
+    else ir.RangeFrame
+  }
+
+  private def buildFrame(ctx: WindowFrameBoundContext): ir.FrameBoundary = {
+    if (Option(ctx.UNBOUNDED()).isDefined) {
+      ir.FrameBoundary(current_row = false, unbounded = true, value = ir.Noop)
+    } else if (Option(ctx.CURRENT()).isDefined) {
+      ir.FrameBoundary(current_row = true, unbounded = false, ir.Noop)
+    } else {
+      Option(ctx.INT())
+        .map(intNode =>
+          ir.FrameBoundary(
+            current_row = false,
+            unbounded = false,
+            value = ir.Literal(integer = Some(intNode.getText.toInt))))
+        .getOrElse(ir.FrameBoundary(current_row = false, unbounded = false, ir.Noop))
+    }
```

Comment:
For consistency with the rest of the code:

```suggestion
    ctx match {
      case c if c.UNBOUNDED() != null => r.FrameBoundary(current_row = false, unbounded = true, value = ir.Noop)
      case c if c.CURRENT() != null => ir.FrameBoundary(current_row = true, unbounded = false, ir.Noop)
      case c if c.INT() != null =>
         ir.FrameBoundary(
            current_row = false,
            unbounded = false,
            value = ir.Literal(integer = Some(c.INT().getText.toInt)))
      case _ => ir.FrameBoundary(current_row = false, unbounded = false, ir.Noop)
    }
```

Reply:
OK - no problem, (well, without the typo)